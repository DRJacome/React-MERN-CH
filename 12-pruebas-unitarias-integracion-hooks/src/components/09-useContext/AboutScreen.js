import React, { useContext } from 'react'
import { Contexto } from './UserContext'

export const AboutScreen = () => {
  const { usuario, setUsuario } = useContext(Contexto)
  const gestionarClick = () => {
    setUsuario({})
  }
  return (
    <div>
      <h1>Acerca de</h1>
      <hr />
      <pre>{JSON.stringify(usuario, null, 3)}</pre>
      <button className='btn btn-primary' onClick={gestionarClick}>Cerrar sesi√≥n</button>
    </div>
  )
}