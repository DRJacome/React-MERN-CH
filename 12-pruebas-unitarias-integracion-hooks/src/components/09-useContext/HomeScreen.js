import React, { useContext } from 'react'
import { Contexto } from './UserContext'

export const HomeScreen = () => {
  const { usuario } = useContext(Contexto)
  console.log(usuario);
  return (
    <div>
      <h1>Pantalla de inicio</h1>
      <hr />
      <pre>{JSON.stringify(usuario, null, 3)}</pre>
    </div>
  )
}