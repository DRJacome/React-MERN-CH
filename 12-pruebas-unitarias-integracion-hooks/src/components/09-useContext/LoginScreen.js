import React, { useContext } from 'react'
import { Contexto } from './UserContext'

export const LoginScreen = () => {
  // 1. Obtener la referencia al userContext.
  const { setUsuario } = useContext(Contexto)
  // 2. setUser.
  /*  {
     id: 123,
     nombre: 'David'
   } */

  return (
    <div>
      <h1>Pantalla de Login</h1>
      <hr />
      <button className='btn btn-primary' onClick={() => {
        setUsuario({
          id: 123,
          nombre: 'David'
        })
      }}>Login</button>
    </div>
  )
}