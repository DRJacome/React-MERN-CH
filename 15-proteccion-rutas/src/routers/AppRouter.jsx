import React, { useContext } from "react";
import {
  BrowserRouter as Router,
  Switch
} from "react-router-dom";
import { VistaLogin } from "../components/login/VistaLogin";
import { PanelRutas } from "./PanelRutas";
import { RutaPrivada } from "./RutaPrivada";
import { RutaPublica } from "./RutaPublica";
import { AutenticacionContext } from "../auth/AutenticacionContext";

export const AppRouter = () => {
  const { usuario } = useContext(AutenticacionContext)
  console.log(usuario);

  return (
    <Router>
      <div>
        <Switch>
          {/* Rutas p√∫blicas */}
          <RutaPublica
            exact path='/login'
            component={VistaLogin}
            isAutenticado={usuario.logged}
          />

          {/* Rutas privadas */}
          <RutaPrivada
            path='/'
            component={PanelRutas}
            isAutenticado={usuario.logged}
          />
        </Switch>
      </div>
    </Router>
  )
}