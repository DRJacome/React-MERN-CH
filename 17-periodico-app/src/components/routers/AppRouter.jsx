import React from 'react'
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect
} from "react-router-dom";
import { AuthRouter } from './AuthRouter';
import { PantallaDiario } from '../diario/PantallaDiario';



export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Switch>
          <Route path='/auth' component={AuthRouter} />
          <Route exact path='/' component={PantallaDiario} />
          <Redirect to='/auth/login' />
        </Switch>
      </div>
    </Router>
  )
}


