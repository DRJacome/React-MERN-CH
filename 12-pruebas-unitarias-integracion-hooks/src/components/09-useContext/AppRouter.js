import React from "react";
/* Para React Router v5.x -> usar Switch;
* Para React Router V6.x -> usar Routes.
 */
import {
  BrowserRouter as Router,
  Switch, //Routes,
  Route
} from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { LoginScreen } from "./LoginScreen";
import { HomeScreen } from "./HomeScreen";
import { NavBar } from "./NavBar";
import { Redirect } from "react-router-dom";


export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <div className="container">
          <Switch>
            <Route exact path="/" component={HomeScreen} />
            <Route exact path="/about" component={AboutScreen} />
            <Route exact path="/login" component={LoginScreen} />
            <Redirect to='/' /> {/* <Route component={HomeScreen} /> */}
          </Switch>
        </div>
      </div>
    </Router>
  )
}