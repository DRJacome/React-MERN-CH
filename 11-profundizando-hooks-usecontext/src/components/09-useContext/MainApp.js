import { useState } from "react"
import { AppRouter } from "./AppRouter"
import { Contexto } from "./UserContext"

export const MainApp = () => {
  const [usuario, setUsuario] = useState({})
  return (
    <Contexto.Provider value={{
      usuario,
      setUsuario
    }}>
      <AppRouter />
    </Contexto.Provider>
  )
}