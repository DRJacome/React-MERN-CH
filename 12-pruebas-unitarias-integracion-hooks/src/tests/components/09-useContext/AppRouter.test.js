import React from 'react'
import { mount } from 'enzyme'
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { Contexto } from '../../../components/09-useContext/UserContext';

describe('Pruebas en <AppRouter/>', () => {

  const usuario = {
    id: 1,
    nombre: 'David'
  }

  const wrapper = mount(
    <Contexto.Provider value={{
      usuario
    }}>
      <AppRouter />
    </Contexto.Provider>
  );

  test('Debe mostrarse correctamente', () => {

    expect(wrapper).toMatchSnapshot();
  })
})