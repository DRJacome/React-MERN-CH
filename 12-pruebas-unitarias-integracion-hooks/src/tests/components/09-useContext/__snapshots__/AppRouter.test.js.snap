// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter/> Debe mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      useContext
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/about"
          onClick={[Function]}
        >
           Acerca de
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        Pantalla de inicio
      </h1>
      <hr />
      <pre>
        {
   "id": 1,
   "nombre": "David"
}
      </pre>
    </div>
  </div>
</div>
`;
